Sources/Widgets/Manipulators/TrackballManipulator/index.js

```diff
@@ -83,15 +100,34 @@ function vtkTrackballManipulator(publicAPI, model) {
 // ----------------------------------------------------------------------------
 
 const DEFAULT_VALUES = {
-  normal: [0, 0, 1],
+  // userOrigin: null,
+  // userNormal: null,
+  // handleOrigin: null,
+  // handleNormal: null,
+  // widgetOrigin: null,
+  // widgetNormal: null
+  useCameraNormal: true,
 };
 
 // ----------------------------------------------------------------------------
 
 export function extend(publicAPI, model, initialValues = {}) {
   Object.assign(model, DEFAULT_VALUES, initialValues);
   macro.obj(publicAPI, model);
-  macro.setGetArray(publicAPI, model, ['normal'], 3);
+  macro.setGet(publicAPI, model, ['useCameraNormal']);
+  macro.setGetArray(
+    publicAPI,
+    model,
+    [
+      'userNormal',
+      'userOrigin',
+      'handleNormal',
+      'handleOrigin',
+      'widgetNormal',
+      'widgetOrigin',
+    ],
+    3
+  );
```

Comment:
looks like you need to create a "base" manipulator that exposes all those models/setters/getters/getOrigin()/getNormal()